src/state/hooks.ts

```diff
@@ -78,13 +78,54 @@ export const useFarmUser = (pid) => {
   }
 }
 
-export const useLpTokenPrice = (symbol: string) => {
-  const farm = useFarmFromSymbol(symbol)
-  const tokenPriceInUsd = useGetApiPrice(getAddress(farm.token.address))
+export const useFarmFromTokenSymbol = (tokenSymbol: string, preferredQuoteTokens?: string[]): Farm => {
+  const farms = useSelector((state: State) => state.farms.data.filter((f) => f.token.symbol === tokenSymbol))
+  const filteredFarm = filterFarmsByQuoteToken(farms, preferredQuoteTokens)
+  return filteredFarm
+}
+
+export const useTokenPriceBusd = (pid: number): BigNumber => {
+  const farm = useFarmFromPid(pid)
+  const bnbPriceBusd = usePriceBnbBusd()
+  const quoteTokenFarm = useFarmFromTokenSymbol(farm.token.symbol)
+
+  if (farm.quoteToken.symbol === 'BUSD') {
+    return farm.tokenPriceVsQuote ? new BigNumber(farm.tokenPriceVsQuote) : BIG_ZERO
+  }
+
+  if (farm.quoteToken.symbol === 'wBNB') {
+    return bnbPriceBusd.gt(0) && bnbPriceBusd.times(farm.tokenPriceVsQuote)
+  }
 
-  return farm.lpTotalSupply && farm.lpTotalInQuoteToken
-    ? new BigNumber(getBalanceNumber(farm.lpTotalSupply)).div(farm.lpTotalInQuoteToken).times(tokenPriceInUsd).times(2)
-    : BIG_ZERO
+  // Possible alternative farm quoteTokens:
+  // UST (MIR-UST), pBTC (PNT-pBTC), BTCB (bBADGER-BTCB), ETH (SUSHI-ETH)
+  // If the farm's quote token isn't BUSD or wBNB, we then use the quote token farm's quote token
+  // i.e. for farm PNT - pBTC
+  // we find the pBTC farm, pBTC - BNB
+  // from the BNB - pBTC BUSD price, we can get the PNT - BUSD price
+
+  if (quoteTokenFarm.quoteToken.symbol === 'wBNB') {
```

Comment:
If the farm doesn't have a quoteToken of BUSD or BNB, we access the quoteToken... of the quoteToken.

For example:
`bBADGER-BTCB`
Find a BTCB  BUSD or BNB farm
When we know the value of BTCB to BUSD or BNB, we can find the value of bBADGER to BUSD